.github/workflows/rust-mirai.yml

```diff
@@ -0,0 +1,64 @@
+name: Rust
+
+on:
+  pull_request:
+    types: ['opened', 'edited', 'reopened', 'synchronize']
+
+jobs:
+  install_and_run_MIRAI:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Z3
+      uses: pavpanchekha/setup-z3@v1.2
+    - name: Checkout MIRAI
+      uses: actions/checkout@v2
+      with:
+        repository: 'facebookexperimental/MIRAI'
+        path: 'MIRAI'
+    - name: Install MIRAI
+      working-directory: ./MIRAI
+      run: |
+        rustup component add rustc-dev
+        RUSTFLAGS='-Clink-arg=-L./binaries -Clink-arg=-lstdc++' cargo install --path ./checker
+    - name: Checkout libra
+      uses: actions/checkout@v2
+      with:
+        path: 'libra'
+    - name: Build libra master then checkout PR
+      working-directory: ./libra
+      run: |
+        rustup override set `cat ../MIRAI/rust-toolchain`
+        rustup component add rustc-dev
+        git fetch origin master
+        git checkout master
+        RUSTFLAGS="-Z always_encode_mir" cargo build
+        git checkout -
+    - name: Run MIRAI on PR
+      working-directory: ./libra
+      run: |
+        rustup show
+        RUSTC_WRAPPER=mirai RUSTFLAGS="-Z always_encode_mir" cargo build > ../mirai_results
+        cat ../mirai_results
+    - name: Post comment with MIRAI warnings
```

Comment:
I see the results in the checks list, but I do not see comments in your test PR. Does this part actually work? Code looks ok at a glance.

Reply:
I've tested it without MIRAI and it works, but I so far haven't been able to successfully test it with MIRAI. Let's see if I can manage to do it before accepting this PR.

Reply:
Btw, you will probably also want to delete the old comment and add a new on on synchronize events so that we don't spam the issue. See this for an example: https://github.com/metajack/cargo-dep-bot/blob/master/src/index.ts#L43-L49

Reply:
mmm, I don't know if this is a good idea. For example, what if I see a MIRAI issue, then correct the code to see if it is still there. I will still want to see the original problem. If the problem is persisting but has changed, I want to see the change in MIRAI's warning.